fvm/src/kernel/error.rs

```diff
@@ -35,6 +37,16 @@ pub enum ExecutionError {
     Fatal(anyhow::Error),
 }
 
+impl Clone for ExecutionError {
+    fn clone(&self) -> Self {
+        match self {
+            OutOfGas => OutOfGas,
+            Syscall(v) => Syscall(v.clone()),
+            Fatal(v) => Fatal(anyhow::Error::msg(v.to_string())),
+        }
+    }
+}
```

Comment:
Hm... this isn't really a clone. I'd handle these cases separately:

1. fatal/out-of-gas: nothing? We can just cut the trace short because it doesn't really matter.
2. Syscall error -> push that.

Reply:
Basically, record `Result<InvocationResult, SyscallError>` instead of `Result<InvocationResult, ExecutionError>`.